akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.internal
+
+import java.sql.Connection
+
+import akka.Done
+import akka.projection.internal.OffsetSerialization.StorageRepresentation
+
+trait Dialect {
+  val createTableStatement: String
+  val alterTableStatement: String
+
+  val insertOrUpdateStatement: String
+  val readOffsetQuery: String
+  val clearOffsetStatement: String
+
+}
+
+/* Defines the basic statements. Dialects may use it or define their own.
+ */
+object DialectDefaults {
+  def readOffsetQuery(table: String) =
+    s"SELECT * FROM $table WHERE PROJECTION_NAME = ?"
+
+  def clearOffsetStatement(table: String) =
+    s"DELETE FROM $table WHERE projection_name = ? AND projection_key = ?"
+
+  def insertOrUpdateStatement(table: String) =
+    s"""MERGE INTO "$table" ("PROJECTION_NAME","PROJECTION_KEY","OFFSET","MANIFEST","MERGEABLE","LAST_UPDATED")  VALUES (?,?,?,?,?,?)"""
+
+}
+case class H2Dialect(schema: Option[String], tableName: String) extends Dialect {
```

Comment:
Only H2 for the moment. We need to add a few more and configure the build to run dockers for each DB.